src/analysis/retail/monk/mistweaver/modules/spells/RapidDiffusion.tsx

```diff
@@ -0,0 +1,158 @@
+  get totalRemThroughput() {
+    return this.remHealing + this.remAbsorbed;
+  }
+
+  get totalVivifyThroughput() {
+    return this.extraVivHealing + this.extraVivAbsorbed;
+  }
+
+  static dependencies = {
+    hotTracker: HotTrackerMW,
+    combatants: Combatants,
+    mistyPeaks: MistyPeaks,
+    vivify: Vivify,
+  };
+  hotTracker!: HotTrackerMW;
+  combatants!: Combatants;
+  mistyPeaks!: MistyPeaks;
+  vivify!: Vivify;
+  remCount: number = 0;
+  remHealing: number = 0;
+  remAbsorbed: number = 0;
+  remOverHealing: number = 0;
+  extraMistyPeaksProcs: number = 0;
+  extraVivCleaves: number = 0;
+  extraVivHealing: number = 0;
+  extraVivOverhealing: number = 0;
+  extraVivAbsorbed: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.RAPID_DIFFUSION_TALENT);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.handleReMApply,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.handleReMHeal,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY), this.handleVivify);
+  }
+
+  handleReMApply(event: ApplyBuffEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromRapidDiffusion(hot) && !this.hotTracker.fromBounce(hot)) {
+      this.remCount += 1;
+    }
+  }
+
+  handleReMHeal(event: HealEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromRapidDiffusion(hot)) {
+      this.remHealing += event.amount || 0;
+      this.remAbsorbed += event.absorbed || 0;
+      this.remOverHealing += event.overheal || 0;
+    }
+  }
+
+  handleVivify(event: HealEvent) {
+    const targetId = event.targetID;
+    if (
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hasAttribution(hot.attributions, RAPID_DIFFUSION)) {
+      this.extraVivCleaves += 1;
+      this.extraVivHealing += event.amount || 0;
+      this.extraVivOverhealing += event.overheal || 0;
+      this.extraVivAbsorbed += event.absorbed || 0;
+    }
+  }
+
+  hasAttribution(attributions: Attribution[], name: string) {
+    return attributions.some(function (attr) {
+      return attr.name === name;
+    });
+  }
```

Comment:
```suggestion
    if (this.hotTracker.fromRapidDiffusion(hot)) {
      this.extraVivCleaves += 1;
      this.extraVivHealing += event.amount || 0;
      this.extraVivOverhealing += event.overheal || 0;
      this.extraVivAbsorbed += event.absorbed || 0;
    }
  }
```

`hasAttribution` seems unnecessary since we already have `isFromRapidDiffusion` in HotTracker